src/component/spinner.js

```diff
@@ -165,26 +226,48 @@ SpinnerFill.propTypes = {
   color: PropTypes.string.isRequired,
 };
 
-const generateArc = (percentage, radius) => {
-  if (percentage === 0) {
-    percentage = 0.001;
-  } else if (percentage === 1) {
-    percentage = 0.9999;
-  }
-  const a = percentage * 2 * Math.PI; // angle (in radian) depends on percentage
-  const r = radius; // radius of the circle
-  var rx = r,
-    ry = r,
-    xAxisRotation = 0,
-    largeArcFlag = 1,
-    sweepFlag = 1,
-    x = r + r * Math.sin(a),
-    y = r - r * Math.cos(a);
-  if (percentage <= 0.5) {
-    largeArcFlag = 0;
-  } else {
-    largeArcFlag = 1;
-  }
+/**
+ * @typedef {Object} Point
+ * @property {number} x The X Coordinate
+ * @property {number} y The Y Coordinate
+ */
+
+/**
+ * Translate radius + angle information into cartestian x and y
+ * @param  {number} centerX        The X-coord of the center of the circle
+ * @param  {number} centerY        The Y-coord of the center of the circle
+ * @param  {number} radius         The radius of the circle
+ * @param  {number} angleInDegrees The angle at which the point sits
+ * @return {Point}                 The point in cartesian form
+ */
+const polarToCartesian = (centerX, centerY, radius, angleInDegrees) => {
+  const angleInRadians = (angleInDegrees - 90) * Math.PI / 180.0;
+
+  return {
+    x: centerX + radius * Math.cos(angleInRadians),
+    y: centerY + radius * Math.sin(angleInRadians),
+  };
+};
+
+/**
+ * Create an SVG path string for a circular arc
+ * @param  {number} x          The X-coord of the center of the circle
+ * @param  {number} y          The Y-coord of the center of the circle
+ * @param  {number} radius     The radius of the circle
+ * @param  {number} startAngle The angle at which to start the arc
+ * @param  {number} endAngle   The angle at which to end the arc
+ * @return {string}            The path string for the arc
+ */
+const generateArc = (x, y, radius, startAngle, endAngle) => {
```

Comment:
I would be happy to move these functions into a helper file so they are more easily unit tested. I don't really see the point of putting them in a `/computed/spinner` file since they don't currently rely on anything in the store (I know one of the spinner does use `store.percentSynced` but that doesn't touch this code path) and that would just seems like over-engineering to me.

Reply:
> they don't currently rely on anything in the store

That's a valid point. My main concern is readaility and separation of concerns. Until now, we've been strict about keeping JS logic out and keeping the react components dumb. This way we don't need to spin up a dom or do testing on the component level, since all logic is already unit tested at the `computed` and `action` level.

Reply:
+1 for moving this into helper / util file and unit testing it.